### PR TITLE
Import TensorFlow models via ONNX conversion

### PR DESCRIPTION
**Pull request by lesters**
@bratseth Please review. Second attempt at converting TF models to ONNX. The main change since last time is the `vespa-convert-tf2onnx` wrapper script around the direct `tf2onnx` call we did last time in `TensorFlowImporter.java`.

### CODE REVIEW

File: model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java [hunk 1]
```diff
@@ -78,15 +69,18 @@ public ImportedModel importModel(String modelName, String modelDir, SavedModelBu
...
+                log.info("Converting TensorFlow model '" + modelDir + "' to ONNX with opset " + opset + "...");
+                Pair<Integer, String> res = convertToOnnx(modelDir, convertedPath, opset);
+                if (res.getFirst() == 0) {
+                    log.info("Conversion to ONNX with opset " + opset + " successful.");
+                    return onnxImporter.importModel(modelName, convertedPath);
+                }
+                log.info("Conversion to ONNX with opset " + opset + " failed. Reason: " + res.getSecond());
             }
-            return onnxImporter.importModel(modelName, convertedPath);
+            throw new IllegalArgumentException("Unable to convert TensorFlow model in '" + modelDir + "' to ONNX.");
```
Comment (bratseth) [inline model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java:83]: Might be useful to include the last reason here?

File: model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java [hunk 2]
```diff
@@ -78,15 +69,18 @@ public ImportedModel importModel(String modelName, String modelDir, SavedModelBu
...
-                throw new IllegalArgumentException("Conversion from TensorFlow to ONNX failed for '" + modelDir + "'. " +
-                        "Reason: " + res.getSecond());
+            for (int opset : onnxOpsetsToTry) {
+                log.info("Converting TensorFlow model '" + modelDir + "' to ONNX with opset " + opset + "...");
+                Pair<Integer, String> res = convertToOnnx(modelDir, convertedPath, opset);
+                if (res.getFirst() == 0) {
+                    log.info("Conversion to ONNX with opset " + opset + " successful.");
+                    return onnxImporter.importModel(modelName, convertedPath);
+                }
+                log.info("Conversion to ONNX with opset " + opset + " failed. Reason: " + res.getSecond());
```
Comment (bratseth) [inline model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java:81]: I think this should probably be Level.FINE?

**Review by bratseth — APPROVED** [review]
LGTM

File: model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java (hunk 1 above)
Comment (lesters) [reply model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java]: Good suggestion. Fixed.

File: model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java (hunk 2 above)
Comment (lesters) [reply model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java]: Fixed.

**Pull request merged**